Update fails in Zabbix LXC due to wrong agent service

### ✅ Have you read and understood the above guidelines?

yes

### 🔎 Did you run the script with verbose mode enabled?

Yes, verbose mode was enabled and the output is included below

### 📜 What is the name of the script you are using?

zabbix

### 📂 What was the exact command used to execute the script?

https://raw.githubusercontent.com/community-scripts/ProxmoxVE/main/ct/zabbix.sh

### ⚙️ What settings are you using?

- [x] Default Settings
- [x] Advanced Settings

### 🖥️ Which Linux distribution are you using?

debian 13

### 📈 Which Proxmox version are you on?

Virtual Environment 8.4

### 📝 Provide a clear and concise description of the issue.

When I run the update on my LXC using cli update, it fails with this error:
Stopping ServicesFailed to stop zabbix-agent.service: Unit zabbix-agent.service not loaded.
in line 46: exit code 5 (Unknown error)
On my system, only zabbix-agent2.service exists

<img width="1299" height="384" alt="Image" src="https://github.com/user-attachments/assets/34b244b6-80d7-40dc-85d0-3b1cfee75f7a" />

<img width="1312" height="528" alt="Image" src="https://github.com/user-attachments/assets/51f82bf9-8e40-4fac-b789-f7a16d8ac76b" />

### 🔄 Steps to reproduce the issue.

Run the update script

### ❌ Paste the full error output (if available).

cat /tmp/create-lxc-cd7b3a92.log
[2026-02-15 21:05:23] [INFO] Stopping Services
[2026-02-15 21:05:23] [ERROR] in line 46: exit code 5 (Unknown error): while executing command systemctl stop "$AGENT_SERVICE"
[2026-02-15 21:05:23] Full log: /tmp/create-lxc-cd7b3a92.log

### 🖼️ Additional context (optional).

_No response_

--- a/ct/zabbix.sh
+++ b/ct/zabbix.sh
@@ -35,15 +35,18 @@ function update_script() {
     exit
   fi
 
-  if systemctl list-unit-files | grep -q zabbix-agent2.service; then
+  if systemctl cat zabbix-agent2.service &>/dev/null; then
     AGENT_SERVICE="zabbix-agent2"
-  else
+  elif systemctl cat zabbix-agent.service &>/dev/null; then
     AGENT_SERVICE="zabbix-agent"
+  else
+    AGENT_SERVICE=""
+    msg_warn "No Zabbix Agent service found, skipping agent actions"
   fi
 
   msg_info "Stopping Services"
   systemctl stop zabbix-server
-  systemctl stop "$AGENT_SERVICE"
+  [[ -n "$AGENT_SERVICE" ]] && systemctl stop "$AGENT_SERVICE"
   msg_ok "Stopped Services"
 
   read -rp "Choose Zabbix version [1] 7.0 LTS  [2] 7.4 (Latest Stable)  [3] Latest available (default: 2): " ZABBIX_CHOICE
@@ -83,13 +86,13 @@ function update_script() {
 
   $STD apt install --only-upgrade zabbix-server-pgsql zabbix-frontend-php php8.4-pgsql
 
-  if [ "$AGENT_SERVICE" = "zabbix-agent2" ]; then
+  if [[ "$AGENT_SERVICE" == "zabbix-agent2" ]]; then
     $STD apt install --only-upgrade zabbix-agent2 zabbix-agent2-plugin-postgresql
     if [ -f /etc/zabbix/zabbix_agent2.d/plugins.d/nvidia.conf ]; then
       sed -i 's|^Plugins.NVIDIA.System.Path=.*|# Plugins.NVIDIA.System.Path=/usr/libexec/zabbix/zabbix-agent2-plugin-nvidia-gpu|' \
         /etc/zabbix/zabbix_agent2.d/plugins.d/nvidia.conf
     fi
-  else
+  elif [[ "$AGENT_SERVICE" == "zabbix-agent" ]]; then
     $STD apt install --only-upgrade zabbix-agent
   fi
 
@@ -105,7 +108,7 @@ function update_script() {
 
   msg_info "Starting Services"
   systemctl start zabbix-server
-  systemctl start "$AGENT_SERVICE"
+  [[ -n "$AGENT_SERVICE" ]] && systemctl start "$AGENT_SERVICE"
   systemctl restart apache2
   msg_ok "Started Services"
   msg_ok "Updated successfully!"

